### PR TITLE
Attention warning

### PR DESCRIPTION
**Pull request by TimOgden**
I thought it would be wise to add this warning to the tutorial since my carelessness cost me many hours of debugging and confusion.

The issue lied in that this tutorial defines its own `MultiHeadAttention` layer separate from `tf.keras.layers.MultiHeadAttention` which has since been created after the creation of this tutorial. I assume many people will do as I did and think "oh I don't need to use the MultiHeadAttention layer defined here, keras already has one!" and will replace it without thinking.

Unfortunately, there is one major difference between these two layers. Our mha layer defined in the tutorial uses the mask to represent which tokens to **NOT** pay attention to, however the mha layer defined by tf.keras uses the mask to represent which tokens to attend to. In my opinion, the tf.keras representation makes more sense, but there is no harm in how the tutorial uses the mask.

However, using the tf.keras mha layer and the `create_look_ahead_mask` method resulted in my model literally paying attention to every future token and ignoring every token it has already seen. This was resulting in my model attaining a concerning 100% accuracy and near-zero loss quickly, which makes sense because every token in the output sequence needed to be mapped to exactly what it could attend to in the input, which is a pretty trivial task.

Nowhere in this tutorial or in the documentation for `tf.keras.MultiHeadAttention` is there any warning for this and someone who is unfamiliar with attention and Transformers (the audience of this tutorial) would have no reason to question the definition of the mask laid out in the tutorial, so I thought a warning would suffice to prevent any other poor souls from making this mistake.

### CODE REVIEW

File: docs/tutorials/transformer.ipynb [hunk 1]
```diff
@@ -1,2354 +1,2386 @@
...
+    "x = tf.random.uniform((1, 3))\n",
+    "temp = create_look_ahead_mask(x.shape[1])\n",
+    "temp"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note: Take note that this mask indicates which entries to *not* attend to. The newly added MultiHeadAttention layer in Keras considers the invered mask, where 1 is a token to be attended to and 0 should be ignored. If you are using `tf.keras.layers.MultiHeadAttention` instead of the MultiHeadAttention layer in this tutorial defined below, use the following commented out method to generate the mask. You will need to `pip install tensorflow-probability` and `import tensorflow_probability as tfp`."
```
Comment (broken) [inline docs/tutorials/transformer.ipynb:713]: A few minor nit-pics:
1. You can remove the initial "Take note that"
2. In the second sentence, remove "newly added" (since people may read this much later).
3. Missing 't' in inverted. ... Keras considers the inverted mask, ...

**Review by broken — CHANGES_REQUESTED** [review]
Thanks for this update!

File: docs/tutorials/transformer.ipynb (hunk 1 above)
Comment (TimOgden) [reply docs/tutorials/transformer.ipynb]: Good points, pushed a commit to fix these changes. Let me know if there are any other suggestions, thanks!

File: docs/tutorials/transformer.ipynb [hunk 2]
```diff
@@ -710,7 +710,7 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "Note: Take note that this mask indicates which entries to *not* attend to. The newly added MultiHeadAttention layer in Keras considers the invered mask, where 1 is a token to be attended to and 0 should be ignored. If you are using `tf.keras.layers.MultiHeadAttention` instead of the MultiHeadAttention layer in this tutorial defined below, use the following commented out method to generate the mask. You will need to `pip install tensorflow-probability` and `import tensorflow_probability as tfp`."
+    "Note: This mask indicates which entries to *not* attend to. However, the MultiHeadAttention layer in Keras that was added after the creation of this tutorial considers the invered mask, where 1 is a token to be attended to and 0 should be ignored. If you are using `tf.keras.layers.MultiHeadAttention` instead of the MultiHeadAttention layer in this tutorial defined below, use the following commented out method to generate the mask. You will need to `pip install tensorflow-probability` and `import tensorflow_probability as tfp`."
```
Comment (broken) [inline docs/tutorials/transformer.ipynb:713]: Looks good! can you just fix the spelling of inverted in "...considers the invered mask,"

Comment (TimOgden) [reply docs/tutorials/transformer.ipynb]: Change was made, sorry for the delay

**Comment by broken** [comment]
I'm seeing this error message in our internal review tools:

`Notebook to text conversion failed: malformed colab notebook: cell missing required field metadata.id.`

May I ask how you saved this? All the formatting is slightly off, making it look like all the lines have changed, which then makes it difficult to pinpoint the changed lines.

**Comment by TimOgden** [comment]
> I'm seeing this error message in our internal review tools:
> 
> `Notebook to text conversion failed: malformed colab notebook: cell missing required field metadata.id.`
> 
> May I ask how you saved this? All the formatting is slightly off, making it look like all the lines have changed, which then makes it difficult to pinpoint the changed lines.

Hm, I used a local jupyter notebook session to modify it instead of google colab, probably was a dumb move considering who owns Tensorflow haha. I think it would be easiest to close this PR and I'll submit another one with the same note added but through colab.
